codegen/test_data/clients/bar.gogen

```diff
@@ -209,9 +220,7 @@ func (c *barClient) ArgNotStruct(
 	headers map[string]string,
 	r *clientsBarBar.Bar_ArgNotStruct_Args,
 ) (map[string]string, error) {
-	req := zanzibar.NewClientHTTPRequest(
-		c.clientID, "argNotStruct", c.httpClient,
-	)
+	req := c.httpClient.NewClientHTTPRequest("ArgNotStruct", "Bar", "argNotStruct")
```

Comment:
If you define/subscope the method scope at the method definition layer, there would be no need to pass around the client method name, which crosses the abstraction boundaries.

Reply:
I was trying to follow the tchannel client style which has the metrics embedded in the client.

Reply:
Done.